Watchman.Discord/Areas/Users/Services/WarnsService.cs

```diff
@@ -0,0 +1,62 @@
+﻿using Devscord.DiscordFramework.Middlewares.Contexts;
+using Devscord.DiscordFramework.Services;
+using MongoDB.Driver;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Watchman.Cqrs;
+using Watchman.Discord.Areas.Users.BotCommands;
+using Watchman.DomainModel.Warns;
+using Watchman.DomainModel.Warns.Commands;
+using Watchman.DomainModel.Warns.Queries;
+
+namespace Watchman.Discord.Areas.Protection.Services
+{
+    public class WarnsService
+    {
+        private readonly UsersService _usersService;
+        private readonly ICommandBus _commandBus;
+        private readonly IQueryBus _queryBus;
+
+        public WarnsService(UsersService usersService, ICommandBus commandBus, IQueryBus queryBus)
+        {
+            this._usersService = usersService;
+            this._commandBus = commandBus;
+            this._queryBus = queryBus;
+        }
+
+        public Task AddWarnToUser(AddWarnCommand command, Contexts contexts, UserContext targetUser)
+        {
+            var addWarnEventCommand = new AddWarnEventCommand(contexts.User.Id, targetUser.Id, command.Reason, contexts.Server.Id);
+            return this._commandBus.ExecuteAsync(addWarnEventCommand);
+        }
+
+        public async Task<IEnumerable<WarnEvent>> GetWarns(ulong serverId, ulong userId)
+        {
+            var query = new GetWarnEventsQuery(serverId, userId);
+            var response = await this._queryBus.ExecuteAsync(query);
+            return response.WarnEvents;
+        }
+
+        public async Task<string> GetWarnsToString(ulong serverId, ulong userId)
```

Comment:
powinieneś tutaj przekazywać `IEnumerable<WarnEvent>`
bo co jeśli ktoś już wcześniej tamto pobrał i na tym operował
a na koniec chciałby tylko zmienić to na tekst, bez konieczności ponownego pobierania?
w dodatku w tej sytuacji gdyby miało dość filtrowanie, to byłoby niemożliwe wyświetlenie przefiltrowanej już listy warnów

zasada SRP z zasad SOLID o tym mówi

Reply:
racja